src/Microsoft.DotNet.Build.Tasks/PackageFiles/VersionTools.targets

```diff
@@ -44,6 +46,51 @@
     </ItemGroup>
   </Target>
 
+  <!--
+    Read a .gitmodules file and generate updaters based on the contents, if GitModulesPath provided.
+    
+    If a submodule has 'branch = ...' configured with a target remote branch to update to and
+    'sourceBuildAutoUpdate = true', this target creates an updater for it.
+    
+    To restrict this target to only create updaters for a specific submodule, set the
+    'TargetSubmodule' property to the name/path of the submodule.
+  -->
+  <Target Name="CreateDefaultSubmoduleUpdateSteps"
+          Condition="'$(GitModulesPath)' != ''">
+    <ReadGitConfigFile File="$(GitModulesPath)">
+      <Output TaskParameter="SubmoduleConfiguration" ItemName="SubmoduleConfiguration" />
+    </ReadGitConfigFile>
+
+    <ItemGroup>
+      <_UpgradableSubmodule Include="@(SubmoduleConfiguration)"
+                            Condition="'%(SubmoduleConfiguration.Branch)' != '' AND
+                                       '%(SubmoduleConfiguration.SourceBuildAutoUpdate)' == 'true'" />
```

Comment:
I think I'm of the opinion that if you have a "Branch" configured in your .gitmodules, and you are using `UpdateDependencies`, then the submodule should be updated by default.  I would think that generally all the submodules should be updatable normally.  And it would only be a special case where we needed to turn off auto-updates on a submodule.

This way we don't have to put `SourceBuildAutoUpdate=true` on every single submodule.

Reply:
From https://github.com/dotnet/source-build/pull/254#discussion_r146281494, it didn't seem like it was clear enough that adding `branch` was the thing that enables this feature. I'm fine putting it back the way I had it, though. Also, Git config files support comments, which would help in source-build's case.

Reply:
My reasoning is - "what do we expect the majority of usages to be?"  If we expect the majority of usages to not auto-update, then opt-in sounds good. But if we expect the majority of usages to want auto-update, then to me it feels like it should be on by default.

When we have all the submodules patches removed and auto-dependency flow, I would expect every submodule to have auto-update turned on.

Reply:
Agreed, that's why it was that way in the first place. I think I overcorrected towards self-documentation.